apps/test/integration/levelSolutions/applab/formatting_visual.js

```diff
@@ -99,5 +101,42 @@ export default {
         testResult: TestResults.FREE_PLAY
       }
     },
+    {
+      description: "belowVisualization is positioned correctly",
+      editCode: true,
+      xml: '',
+      startHtml: `
+          <div id="divApplab" class="appModern running" tabindex="1" style="width: 320px; height: 450px;">
+            <div class="screen" tabindex="1" id="screen1" style=
+            "display: block; height: 450px; width: 320px; left: 0px; top: 0px; position: absolute; z-index: 0;">
+            </div>
+          </div>
+          <div id="designModeViz" class="appModern withCrosshair clip-content" data-radium="true"
+          style="width: 320px; height: 450px; display: block;">
+            <div class="screen" tabindex="1" id="design_screen1" style=
+            "display: block; height: 450px; width: 320px; left: 0px; top: 0px; position: absolute; z-index: 0;">
+            </div>
+          </div>`,
+      runBeforeClick: function (assert) {
+        // Delay this check because the belowVisualization resize doesn't fire
+        // immediately.
+        tickWrapper.runOnAppTick(Applab, 10, function () {
+          // Note: This test is fragile, and we're fine with that - we want to
+          // notice when we make a change that affects the #belowVisualization
+          // area, which is invisible in many scenarios.
+          // See https://github.com/code-dot-org/code-dot-org/pull/15848
+          // as an example of a regression that this would catch.
+          const belowVisualization = document.getElementById('belowVisualization');
+          assert.equal(belowVisualization.style.top, '1073px');
+          assert.equal(belowVisualization.style.bottom, '0px');
+
+          Applab.onPuzzleComplete();
+        });
```

Comment:
I went with a level test for now because the `resizePinnedBelowVisualizationArea` function has a bunch of dependencies on DOM references and jQuery size queries making it difficult to unit test without some dangerous changes.  I'm still going to revisit that, but suspect these more straightforward tests should go in on their own - and they'll help catch any mistakes we make with later refactoring.